.github/workflows/ci.yml

```diff
@@ -83,8 +92,8 @@ jobs:
               run: python setup.py sdist
 
             - uses: actions/upload-artifact@v4
-              if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags/v')
```

Comment:
The upload destination is specific for a given run of the workflow. So we always want to upload to check that things are working (previously they weren't working because all the uploads had the same name).